pydantic/_internal/_generate_schema.py

```diff
@@ -1307,10 +1318,7 @@ def set_discriminator(schema: CoreSchema) -> CoreSchema:
         this_field_validators = [v for v in this_field_validators if v not in each_item_validators]
         schema = apply_each_item_validators(schema, each_item_validators, name)
 
-        schema = apply_validators(schema, filter_field_decorator_info_by_field(this_field_validators, name), name)
```

Comment:
Note that this line was also modified, the `field_field_decorator_info_by_field` call was removed as it was already done a few lines above.